pkg/apis/core/v1/defaults.go

```diff
@@ -349,6 +349,21 @@ func SetDefaults_HTTPGetAction(obj *v1.HTTPGetAction) {
 		obj.Scheme = v1.URISchemeHTTP
 	}
 }
+
+// SetDefaults_Namespace adds a default label for all namespaces
+func SetDefaults_Namespace(obj *v1.Namespace) {
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.DefaultNamespaceLabelStamping) {
+		if obj.Labels == nil {
+			obj.Labels = map[string]string{}
+		}
+		// Default label that matches the name of the namespace, for selecting w/o apriori knowledge of labels.
+		// Update to this label is not allowed. Hence, the value of the label is intentionally overridden here
+		// with the namespace name.
+		obj.Labels["kubernetes.io/metadata.name"] = obj.Name
```

Comment:
I would be in favor of putting `kubernetes.io/metadata.name` in a constant, ideally in a staging repo for external consumers.

Reply:
i was thinking of the same.. but i am not sure where such constants need to go

Reply:
This is probably the best place https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/well_known_labels.go

Reply:
added `LabelNamespaceName` pushing shortly.

Reply:
pusheeeed